diff -rupN u-boot.orig/cmd/hwrev.c u-boot/cmd/hwrev.c
--- u-boot.orig/cmd/hwrev.c	2025-05-31 22:49:00.193557511 +0000
+++ u-boot/cmd/hwrev.c	2025-05-31 22:51:17.817163331 +0000
@@ -19,25 +19,15 @@ int do_hwrev(cmd_tbl_t *cmdtp, int flag,
 		return CMD_RET_FAILURE;
 	}
 
-	/* GO2 rev 1.1 */
-	if (check_range(655, 695, hwrev_adc)) {
-		env_set("hwrev", "v11");
-		env_set("dtb_name", "rk3326-odroidgo2-linux-v11.dtb");
-	}
-	/* GO2 rev 1.0 */
-	else if (check_range(816, 896, hwrev_adc)) {
-		env_set("hwrev", "v10");
-		env_set("dtb_name", "rk3326-odroidgo2-linux.dtb");
-	}
-	/* GO3 rev 1.0 */
-	else if (check_range(40, 126, hwrev_adc)) {
-		env_set("hwrev", "v10-go3");
-		env_set("dtb_name", "rk3326-odroidgo3-linux.dtb");
-	}
-	/* engineer samples */
-	else {
-		env_set("hwrev", "v00");
-		env_set("dtb_name", "rk3326-odroidgo2-linux.dtb");
+	if (check_range(490, 540, hwrev_adc)) {
+                env_set("hwrev", "k36");
+                env_set("dtb_name", "rk3326-k36-clone.dtb");
+	} else if(check_range(140, 190, hwrev_adc)) {
+                env_set("hwrev", "r33s");
+                env_set("dtb_name", "rk3326-gameconsole-r33s.dtb");
+ 	} else if(check_range(1000, 1050, hwrev_adc)) {
+                env_set("hwrev", "xumini");
+                env_set("dtb_name", "rk3326-magicx-xu-mini-m.dtb");
	}
 
 	printf("adc0 (hw rev) %d\n", hwrev_adc);
