diff --git a/scripts/mkmultidtb.py b/scripts/mkmultidtb.py
index 291a223a7..73c86566c 100755
--- a/scripts/mkmultidtb.py
+++ b/scripts/mkmultidtb.py
@@ -26,15 +26,7 @@ DTBS['RK3308-EVB'] = OrderedDict([('rk3308-evb-dmic-i2s-v10', '#_saradc_ch3=288'
 				  ('rk3308-evb-dmic-pdm-v10', '#_saradc_ch3=1024'),
 				  ('rk3308-evb-amic-v10', '#_saradc_ch3=407')])
 
-DTBS['rk356x'] = OrderedDict([('rk3566-rg353p-linux', '#_saradc_ch1=852'),
-			      ('rk3566-rg353m-linux', '#_saradc_ch1=512'),
-			      ('rk3566-rg353v-linux', '#_saradc_ch1=681'),
-			      ('rk3566-rg-arc-linux', '#_saradc_ch1=183'),
-			      ('rk3566-rk2023-linux', '#_saradc_ch1=650'),
-			      ('rk3566-rgb30-linux', '#_saradc_ch1=383'),
-			      ('rk3566-rgb20pro-linux', '#_saradc_ch1=245'),
-			      ('rk3566-max3pro-linux', '#_saradc_ch1=750'),
-			      ('rk3566-rg503-linux', '#_saradc_ch1=1023')])
+DTBS['rk356x'] = OrderedDict([('rk3566-rg-arc-linux', '#_saradc_ch1=183')])
 
 DTBS['rk3588'] = OrderedDict([('rk3588s-orangepi-5', '#_saradc_ch1=001'),
                               ('rk3588-rock-5b', '#_saradc_ch1=002'),
