diff --git a/scripts/mkmultidtb.py b/scripts/mkmultidtb.py
index 291a223a7..e1fd081a3 100755
--- a/scripts/mkmultidtb.py
+++ b/scripts/mkmultidtb.py
@@ -29,7 +29,6 @@ DTBS['RK3308-EVB'] = OrderedDict([('rk3308-evb-dmic-i2s-v10', '#_saradc_ch3=288'
 DTBS['rk356x'] = OrderedDict([('rk3566-rg353p-linux', '#_saradc_ch1=852'),
 			      ('rk3566-rg353m-linux', '#_saradc_ch1=512'),
 			      ('rk3566-rg353v-linux', '#_saradc_ch1=681'),
-			      ('rk3566-rg-arc-linux', '#_saradc_ch1=183'),
 			      ('rk3566-rk2023-linux', '#_saradc_ch1=650'),
 			      ('rk3566-rgb30-linux', '#_saradc_ch1=383'),
 			      ('rk3566-rgb20pro-linux', '#_saradc_ch1=245'),
